src/TerraformApplyCommand.groovy

```diff
@@ -95,34 +89,13 @@ class TerraformApplyCommand implements TerraformCommand, Resettable {
         return pieces.join(' ')
     }
 
-    private applyPluginsOnce() {
-        def remainingPlugins = plugins - appliedPlugins
-
-        for (TerraformApplyCommandPlugin plugin in remainingPlugins) {
-            plugin.apply(this)
-            appliedPlugins << plugin
-        }
-    }
-
-    public static void addPlugin(TerraformApplyCommandPlugin plugin) {
-        plugins << plugin
-    }
-
     public static TerraformApplyCommand instanceFor(String environment) {
         return new TerraformApplyCommand(environment)
             .withInput(false)
             .withArgument("-auto-approve")
     }
 
-    public static getPlugins() {
-        return plugins
-    }
-
     public static reset() {
         this.plugins = []
     }
```

Comment:
Interesting - I never thought about reset.  It seems like `reset()` could be pulled into the Pluggable interface - resetting the plugins is a fundamental part of being Pluggable.  Thoughts?

Reply:
i was a bit confused about Resettable - I moved it around the trait hierarchy a bit and then left it out... is it JUST for plugins? It also seems like it should reset default values as well.

Reply:
No, I would not say that Resettable is JUST for plugins.  

I would say that being Pluggable automatically makes you Resettable - in so far as Plugins are stored as static variables, and Resettable is all about restoring static state as though it had never been modified (during testing).